src/ambulance_game/markov/waiting.py

```diff
@@ -0,0 +1,340 @@
+                    mu,
+                    num_of_servers,
+                    threshold,
+                    system_capacity,
+                    parking_capacity,
+                )
+                mean_waiting_time += current_state_wait * pi[u, v]
+                probability_of_accepting += pi[u, v]
+        mean_waiting_time /= probability_of_accepting
+
+    if formula == "closed_form":
+        sojourn_time = 1 / (num_of_servers * mu)
+        if patient_type == "others":
+            mean_waiting_time = np.sum(
+                [
+                    (state[1] - num_of_servers + 1) * pi[state] * sojourn_time
+                    for state in all_states
+                    if is_accepting_state(
+                        state,
+                        patient_type,
+                        threshold,
+                        system_capacity,
+                        parking_capacity,
+                    )
+                    and state[1] >= num_of_servers
+                ]
+            ) / np.sum(
+                [
+                    pi[state]
+                    for state in all_states
+                    if is_accepting_state(
+                        state,
+                        patient_type,
+                        threshold,
+                        system_capacity,
+                        parking_capacity,
+                    )
+                ]
+            )
+        if patient_type == "ambulance":
+            mean_waiting_time = np.sum(
+                [
+                    (min(state[1] + 1, threshold) - num_of_servers)
+                    * pi[state]
+                    * sojourn_time
+                    for state in all_states
+                    if is_accepting_state(
+                        state,
+                        patient_type,
+                        threshold,
+                        system_capacity,
+                        parking_capacity,
+                    )
+                    and min(state[1], threshold) >= num_of_servers
+                ]
+            ) / np.sum(
+                [
+                    pi[state]
+                    for state in all_states
+                    if is_accepting_state(
+                        state,
+                        patient_type,
+                        threshold,
+                        system_capacity,
+                        parking_capacity,
+                    )
+                ]
+            )
+
+    return mean_waiting_time
+
+
+def get_mean_waiting_time_markov(
+    lambda_a,
+    lambda_o,
+    mu,
+    num_of_servers,
+    threshold,
+    system_capacity,
+    parking_capacity,
+    output="both",
+    formula="closed_form",
+):
+    """Gets the mean waiting time for a Markov chain model
+
+    Parameters
+    ----------
+    lambda_a : float
+    lambda_o : float
+    mu : float
+    num_of_servers : int
+    threshold : int
+    system_capacity : int
+    parking_capacity : int
+    output : str, optional
```

Comment:
I don't understand what this `output` variable is for.

Note that in a few cases here I think we should actually just code two different functions and modularise some of the code used by those functions. Having str switches as variables is good to avoid if possible. This is different to passing a specific python object (like a function) as a variable.

Reply:
That should have been `patient_type`